addons/calendar/tests/test_event_recurrence.py

```diff
@@ -352,6 +352,26 @@ def test_backward_pass_dst(self):
             (datetime(2023, 4, 27, 7, 00), datetime(2023, 4, 27, 8, 00)),
         ])
 
+    def test_all_day_date(self):
+        recurrence = self.env['calendar.event'].with_context(
+            default_start_date=date(2019, 10, 22),
+            default_stop_date=date(2019, 10, 22),
```

Comment:
I would add a default_start ad default_stop relevant. See the parameters of the RPC in the interface to have a test as close as possible to the current behavior.